channels/OFF-CHAIN.md

```diff
@@ -0,0 +1,503 @@
+confused if updates are being proposed simultaneously.
+On a higher level, to keep off chain and on chain state in sync, peers should
+refuse to sign updates for either without also getting a signature for the
+other, e.g. don't sign an on chain `channel_deposit` transaction without also
+updating the state in channel as well.
+
+With the consistency of state updates being secured between peers, it is
+essential, that peers make absolutely sure to not lose their local state, since
+it could potentially lead to them not being able to slash outdated states
+published on chain.
+
+
+## Control messages
+
+
+### `error`
+
+Explicitly communicating errors should make debugging easier. In order to avoid
+complex error handling, which tends to be prone to mistakes, once a channel
+returns an error, it MUST be considered unusable. As such, errors SHOULD be
+reserved for unrecoverable failures only.
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| length               | 2  |
+ ---------------------- ----
+| data                 |    |
+ ---------------------- ----
+```
+
+- `channel_id`: the temporary or final channel id
+- `length`: length of data field
+- (optional) `data`: relevant information
+
+
+### `ping`/`pong`
+
+
+## Establishing channel off-chain
+
+```
+A (initiator)          B
+|                      |
+|---  channel_open  -->|
+|<-- channel_accept ---|
+|                      |
+|-- funding_created -->|
+|<- funding_signed  ---|
+|                      |
+|--- funding_locked -->|
+|<-- funding_locked ---|
+|                      |
+```
+
+In order to establish a channel both peers need to agree on the initial
+conditions, e.g. funding amounts, minimum reserve and lock time.
+
+By default the initiator of the channel will pay the fee for the opening
+transaction.
+
+
+### `channel_open`
+
+This message initiates the opening of a channel and communicates the initiators'
+intent to a potential future peer.
+
+The `channel_open` message should provide the accepting peer all the information
+it needs to assess whether it should accept the channel or not.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| chain_hash           | 32 |
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| lock_period          | 2  |
+ ---------------------- ----
+| push_amount          | 8  |
+ ---------------------- ----
+| initiator_amount     | 8  |
+ ---------------------- ----
+| responder_amount     | 8  |
+ ---------------------- ----
+| channel_reserve      | 8  |
+ ---------------------- ----
+| initiator_pubkey     | 32 |
+ ---------------------- ----
+```
+
+- `chain_hash`: transaction hash of the chain you want to use, e.g. hash of the
+  genesis
+- `temporary_channel_id`: randomly chosen id unique between the involved peers
```

Comment:
Is there a suggested approach for generation of this id for reducing likelihood of tx not applied for conflict on chain?

Reply:
The temporary channel id is part of the channel opening sequence of events. It is picked by the `initiator` and send to the `responder`. It is not part of the create transaction and there is no chance of chain conflict. Once the connection is opened another _real_ channel id is picked and used